azkaban-common/src/main/java/azkaban/imagemgmt/rampup/ImageRampupManagerImpl.java

```diff
@@ -0,0 +1,190 @@
+ * is the version selection process for an image type - 1. Fetch the rampup details for the given
+ * image types or for all the image types (Two such methods are provided). 2. Sort the ramp up data
+ * in ascending order of rampup percentage. 3. Generate a random number between 1 to 100 both
+ * inclusive. Let us say the number the number generated is 60. 4. Let us say there are three
+ * versions 1.1.1, 1.1.2 & and 1.1.3 with rampup percantages 10, 30 and 60 respectively. 5. The
+ * above percentage fors three ranges [1 - 10], [11 - 40] & [41 - 100]. The random humber 60 belongs
+ * to the last range i.e. [41 - 100] and hence version 1.1.3 will be selected. If random number
+ * generated is 22 then version 1.1.2 will be selected and so on. 6. If there is no active rampup
+ * plan for an image type or in the active plan if the version is marked unstable or deprecated, and
+ * latest active version will be selected from the image_verions table. 7. If there is no active
+ * version in the image_versions table, it will throw appropriate error message mentioning could not
+ * select version for the image type and the whole process would fail. 8. Follow the rampup
+ * procedure to elect a new version from the image_versions table for the failed image type.
+ */
+@Singleton
+public class ImageRampupManagerImpl implements ImageRampupManger {
+
+  private static final Logger log = LoggerFactory.getLogger(ImageRampupManagerImpl.class);
+  private final ImageTypeDao imageTypeDao;
+  private final ImageVersionDao imageVersionDao;
+  private final ImageRampupDao imageRampupDao;
+
+  @Inject
+  public ImageRampupManagerImpl(final ImageRampupDao imageRampupDao,
+      final ImageVersionDao imageVersionDao,
+      final ImageTypeDao imageTypeDao) {
+    this.imageRampupDao = imageRampupDao;
+    this.imageVersionDao = imageVersionDao;
+    this.imageTypeDao = imageTypeDao;
+  }
+
+  @Override
+  public Map<String, String> fetchAllImageTypesVersion() throws ImageMgmtException {
+    Map<String, List<ImageRampup>> imageTypeRampups = imageRampupDao.fetchAllImageTypesRampup();
+    List<ImageType> imageTypeList = imageTypeDao.getAllImageTypes();
+    Set<String> imageTypes = new TreeSet<>();
+    for (ImageType imageType : imageTypeList) {
+      imageTypes.add(imageType.getName());
+    }
+    return this.processAndGetVersionForImageTypes(imageTypes, imageTypeRampups);
+  }
+
+  @Override
+  public Map<String, String> fetchVersionByImageTypes(Set<String> imageTypes)
+      throws ImageMgmtException {
+    Map<String, List<ImageRampup>> imageTypeRampups = imageRampupDao
+        .fetchRampupByImageTypes(imageTypes);
+    return this.processAndGetVersionForImageTypes(imageTypes, imageTypeRampups);
+  }
+
+  /**
+   * This method processes image type rampup details for the image type and selects a version for
+   * the image type. Here is the version selection process for an image type 1. Sort the ramp up
```

Comment:
Nit: Add numbering in new line Easy to read.

Reply:
Tried multiple times earlier. But formatting is changing that.